### DISCUSSION TITLE
Face Detection and License Plate Recognition with frigate

### DISCUSSION THREAD

**Discussion by sfilinto** [blakeblackshear/frigate#3842] [General]
New to frigate and impressed with its capabilities with regards to object detection. I don't have a TPU as yes but hoping I will shortly.

Are there any addons / plugins to do facial recognition and also License Plate Recognition & send out alerts around what is detected ?
This will be a small setup in a quiet location and I don't expect more then 10-15 Face Recognitions & even lesser License Plate Recognitions per day.

TIA

**Comment by NickM-27** [comment]
Check out https://github.com/jakowenko/double-take

Frigate+ will have license plate detection: https://frigate.video/plus

**Reply by imne** [reply]
does frigate+ currently have license plate recognition?

**Reply by NickM-27** [reply]
there is a difference between license plate detection and reading. License plate detection is supported as a label that is detected, license plate reading will need to be done via an ALPR software like CodeProjectAI

**Comment by DrSpaldo** [comment]
> there is a difference between license plate detection and reading. License plate detection is supported as a label that is detected, license plate reading will need to be done via an ALPR software like CodeProjectAI

Are you aware of any projects that integrate CodeProjectAI's LPR with Frigate?

**Reply by NickM-27** [reply]
I have a script that does this, not aware of anything cleanly packaged like doubletake

**Reply by JohnNeville** [reply]
I have been using this which does that https://github.com/ljmerza/frigate_plate_recognizer

My camera isn't in a great position for it to work consistently but it has worked great when it catches me pulling into the driveway

**Reply by Wildcat1** [reply]
> I have been using this which does that https://github.com/ljmerza/frigate_plate_recognizer
> 
> My camera isn't in a great position for it to work consistently but it has worked great when it catches me pulling into the driveway

@JohnNeville are you running this locally (Snapshot SDK) or through their Snapshot Cloud ?

**Reply by JohnNeville** [reply]
@Wildcat1 I am using the [code project integration](https://github.com/ljmerza/frigate_plate_recognizer#:~:text=If%20you're%20using%20CodeProject.AI%2C).

**Reply by Wildcat1** [reply]
Ah Ok. I can run with Codeproject but unfortunately that doesn’t appear to support OpenVino therefore crazy high inference times :( hence looking to run Plate Organizer locally.

**Reply by JohnNeville** [reply]
@Wildcat1 this is a distinct usage of code project ai other than what Frigate can use internally.

I am still using my Google coral with the Frigate+ model to detect that a car with a license plate exists in a snapshot. The plate recognizer external container is then effectively just processing that Mqtt event queue and the provided snapshot and then using the Frigate api to enrich that event with extra sublables.

Within Codeproject ai this is the License Plate Reader module and it is just rerunning object detection on that single frame to extract the license plate area from the frame and running OCR on the text. For me it is just running on my CPU because it is only processing individual snapshots that the coral had prescreened to confirm that a license plate exists to read.

**Comment by divemasterjm** [comment]
> I have a script that does this, not aware of anything cleanly packaged like doubletake

can you share it?
thanks

**Reply by NickM-27** [reply]
I am planning to do a discussion post as a guide at some point, it is a bit involved.

**Reply by Sceptersax** [reply]
Also waiting on this. I need a license plate reader. I was hoping to incorporate Codeproject but have no idea on how to make that work with frigate. Since i never used it before, i started with trying it out on blue Iris since I read that they are working together. I tried using codeproject from a docker container, but having problems with GPU integration along with other problems... I'd much rather use it with frigate.

**Reply by idontcare99999** [reply]
Mark me down as another interested party.

**Reply by Terence-T3** [reply]
> I am planning to do a discussion post as a guide at some point, it is a bit involved.

Have you managed to post this yet?

**Reply by NickM-27** [reply]
https://github.com/blakeblackshear/frigate/discussions/8165

**Comment by divemasterjm** [comment]
> I am planning to do a discussion post as a guide at some point, it is a bit involved.

Thanks will wait till then.
